unison-cli/src/Unison/LSP/Queries.hs

```diff
@@ -0,0 +1,146 @@
+-- | Rewrites of some codebase queries, but which check the scratch file for info first.
+module Unison.LSP.Queries
```

Comment:
This module consists of combinators used in #3637 , since these are the combinators which require the Anns to be correct they also turn out to be good testing helpers 😄